system/Helpers/url_helper.php

```diff
@@ -35,123 +35,542 @@
  * @since	Version 3.0.0
  * @filesource
  */
-
-
-if (! function_exists('site_url'))
+if ( ! function_exists('site_url'))
 {
+
 	/**
 	 * Return a site URL to use in views
 	 *
-	 * @param string      $path
+	 * @param string|array      $path
 	 * @param string|null $scheme
+	 * @param \Config\App|null	$altConfig	Alternate configuration to use
 	 *
 	 * @return string
 	 */
-	function site_url(string $path = '', string $scheme = null): string
+	function site_url($path = '', string $scheme = null, \Config\App $altConfig = null): string
 	{
-		$config = new \Config\App();
+		// convert segment array to string
+		if (is_array($path)) $path = implode('/', $path);
+
+		// use alternate config if provided, else default one
+		$config = empty($altConfig) ? new \Config\App() : $altConfig;
 
 		$base = base_url();
 
-		// Add index page
-		if (! empty($config->indexPage))
+		// Add index page, if so configured
+		if ( ! empty($config->indexPage))
 		{
 			$path = rtrim($base, '/').'/'.rtrim($config->indexPage, '/').'/'.$path;
 		}
+		else $path = rtrim($base, '/').'/'.$path;
 
 		$url = new \CodeIgniter\HTTP\URI($path);
 
-		if (! empty($scheme))
+		// allow the scheme to be over-ridden; else, use default
+		if ( ! empty($scheme))
 		{
 			$url->setScheme($scheme);
 		}
 
-		return (string)$url;
+		return (string) $url;
 	}
+
 }
 
 //--------------------------------------------------------------------
 
-if (! function_exists('base_url'))
+if ( ! function_exists('base_url'))
 {
+
 	/**
 	 * Return the base URL to use in views
 	 *
-	 * @param string $path
+	 * @param string|array $path
 	 * @param string $scheme
 	 * @return string
 	 */
 	function base_url(string $path = '', string $scheme = null): string
```

Comment:
Comment was changed to indicate string|array for $path, but function definition is still `string $path`.